Automattic/wp-calypso
# Jetpack Onboarding: State for receiving onboarding credentials

This introduces a basic action and reducer for receiving the onboarding credentials from a site. We need those in order to perform onboarding requests on a site.

To test:
* Checkout this branch on your local Calypso.
* Checkout https://github.com/Automattic/jetpack/pull/8352 on your Jetpack sandbox.
* Apply D7342-code on your wp.com sandbox and make sure you're sandboxing the API and jetpack.wordpress.com.
* Go to https://YourJetpackSandbox.com/wp-admin/admin.php?page=jetpack&action=onboard&calypso_env=development where `YourJetpackSandbox.com` is the domain of your Jetpack sandbox.
* Verify that after a couple of redirects you land in http://calypso.localhost:3000/jetpack/onboarding
* Load Redux DevTools and verify that a `JETPACK_ONBOARDING_CREDENTIALS_RECEIVE` action was dispatched, with a numeric `siteId` and a `credentials`, which contains a `token`, `userEmail` and `siteUrl`.
* Verify the state now contains those credentials (you can do that by typing `state.jetpackOnboarding.credentials` in your console)
* Verify all tests pass: `npm run test-client jetpack-onboarding`
